api/src/main/java/org/eclipse/microprofile/health/Liveness.java

```diff
@@ -38,6 +44,7 @@
 @Qualifier
 @Documented
 @Retention(RetentionPolicy.RUNTIME)
+@Target({TYPE, METHOD, PARAMETER, FIELD})
```

Comment:
I thought you would only permit this annotation on the class level. Does it make sense to be on method or parameter or field at all?

Reply:
We do support CDI Integration for the annotations, such as https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/java-api.asciidoc#integration-with-cdi, where we add the Health check annotation at the method level, in addition to `@Inject`. For the field level, we do use it in the SmallRye impl, as follows https://github.com/smallrye/smallrye-health/blob/16222660d83eeb77bc2f3b2e4c7298be4fb9302e/implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java#L63

Do you know think this should be only used at the class level, instead?

Reply:
CDI producers can be methods or fields - https://docs.oracle.com/javaee/6/tutorial/doc/gkgkv.html. PARAMETER is for injection of resources in implementations I suppose.

Reply:
@pgunapal @xstefank I understand the annotations can be on these places. I was just questioning how useful it is. If both of you think it is useful, I am fine with that. By the way, do you have TCKs for all of the changes? If not, please add some to ensure the implementations obey this.

Reply:
@Emily-Jiang good idea with the TCK, I raised #293 on your behalf. I, however, don't think it's necessary to block now the 3.1 release for these additional tests. But maybe if we still would have time I would like to add them at least for methods. I am not sure if we need to/want to test field or parameter injections as these are for the consuming probably mostly in the implementations. I would like to release RC2 today and the Final tomorrow. I'll see if we can get these tests done in time.

Reply:
Sounds good! Thanks @xstefank !